File: src/python/flask/app/models/health.py
from dataclasses import dataclass
from typing import Optional

@dataclass
class HealthStatus:
    status: str
    version: str
    timestamp: str
    database: str
    cache: str

    @classmethod
    def create(cls, db_healthy: bool, cache_healthy: bool):
        return cls(
            status='healthy' if (db_healthy and cache_healthy) else 'unhealthy',
            version='1.0.0',
            timestamp=datetime.utcnow().isoformat(),
            database='healthy' if db_healthy else 'unhealthy',
            cache='healthy' if cache_healthy else 'unhealthy'
        )

    def to_dict(self):
        return {
            'status': self.status,
            'version': self.version,
            'timestamp': self.timestamp,
            'database': self.database,
            'cache': self.cache
        }

File: src/deno/oak/routes/health.ts
import { Router, Context } from "../deps.ts";
import { databaseService } from "../services/database.ts";
import { cacheService } from "../services/cache.ts";
import { HealthStatus } from "../types.ts";

const router = new Router({
  base: "/health",
});

router.get("/", async (ctx: Context) => {
  const dbHealthy = await databaseService.healthCheck();
  const cacheHealthy = await cacheService.ping();

  const health: HealthStatus = {
    status: dbHealthy && cacheHealthy ? "healthy" : "unhealthy",
    version: "1.0.0",
    timestamp: new Date().toISOString(),
    database: dbHealthy ? "healthy" : "unhealthy",
    cache: cacheHealthy ? "healthy" : "unhealthy",
  };

  ctx.response.status = dbHealthy && cacheHealthy ? 200 : 503;
  ctx.response.headers.set("Content-Type", "application/json");
  ctx.response.body = JSON.stringify(health);
});

router.get("/healthz", (ctx: Context) => {
  ctx.response.body = { status: "ok" };
});

export { router as healthRouter };
